db/models/users.js

```diff
@@ -0,0 +1,77 @@
+const client = require('./index');
+const bcrypt = require('bcrypt');
+const SALT_ROUNDS = 10;
+
+const createUser = async ({ username, password }) => {
+    const hashPwd = await bcrypt.hash(password, SALT_ROUNDS);
+    try {
+        const { rows: [user] } = await client.query(`
+            INSERT INTO users (username, email, password)
+            VALUES ($1, $2, $3)
+            RETURNING *;
+        `, [username, email, hashPwd]);
+        delete user.password
+        return user;
+    } catch (error) {
+        throw error;
+    }
+}
```

Comment:
I'd recommend thinking about what guard clauses we can add to these statements. For example, if the username OR the password are undefined, we can throw a custom error to address that.
`throw {
  name: "FieldsRequired"
  message: "You must provide both a username and password"
}`

these kinds of guard clauses can be helpful once we go to building our front end by giving us very clear and detailed error messages for what went wrong.

Reply:
You got it. I'll work on that now.